### PR TITLE
Display correct Recipient in Sent & Draft folders

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/models/message/Message.kt [hunk 1]
```diff
@@ -149,10 +150,13 @@ class Message : RealmObject {
...
-    inline val sender
-        get() = from.firstOrNull().also {
-            if (it == null) SentryLog.e("ThreadAdapter", "Message $uid has empty from")
+    fun sender(folderRole: FolderRole? = null): Recipient? {
+        val recipients = when (folderRole ?: folder.role) {
+            FolderRole.SENT, FolderRole.DRAFT -> to
+            else -> from
         }
+        return recipients.firstOrNull().also { if (it == null) SentryLog.e("ThreadAdapter", "Message $uid has empty from") }
+    }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/data/models/message/Message.kt:159]: Shouldn't we update the sentry log also ?

File: app/src/main/java/com/infomaniak/mail/data/models/message/Message.kt [hunk 2]
```diff
@@ -149,10 +150,13 @@ class Message : RealmObject {
...
 
-    inline val sender
-        get() = from.firstOrNull().also {
-            if (it == null) SentryLog.e("ThreadAdapter", "Message $uid has empty from")
+    fun sender(folderRole: FolderRole? = null): Recipient? {
+        val recipients = when (folderRole ?: folder.role) {
+            FolderRole.SENT, FolderRole.DRAFT -> to
+            else -> from
         }
+        return recipients.firstOrNull().also { if (it == null) SentryLog.e("ThreadAdapter", "Message $uid has empty from") }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/data/models/message/Message.kt:158]:
```suggestion

        return recipients.firstOrNull().also { if (it == null) SentryLog.e("ThreadAdapter", "Message $uid has empty from") }
```

File: app/src/main/java/com/infomaniak/mail/data/models/thread/Thread.kt [hunk 3]
```diff
@@ -211,27 +211,27 @@ class Thread : RealmObject {
...
+            scope.setExtra("thread.folder.name", folder.name)
+            scope.setExtra("thread.folder.id", folder.id)
+            scope.setExtra("thread.uid", uid)
+            scope.setExtra("thread.messages.count", "${messages.count()}")
+            scope.setExtra("thread.duplicates.count", "${duplicates.count()}")
+            scope.setExtra("thread.isFromSearch", "$isFromSearch")
+            scope.setExtra("thread.hasDrafts", "$hasDrafts")
+            Sentry.captureException(throwable)
         }
+        return@getOrElse null
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/data/models/thread/Thread.kt:229]:
```suggestion

        return@getOrElse null
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 4]
```diff
@@ -83,12 +83,15 @@ class ThreadFragment : Fragment() {
     @Inject
     lateinit var localSettings: LocalSettings
 
+    @Inject
+    lateinit var injectedThreadAdapter: ThreadAdapter
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:87]: Why not just threadAdapter ?

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 4 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:87]: Because there is already a `threadAdapter` here.

**Review by FabianDevel — APPROVED** [review]

**Pull request merged**